leakcanary-android-instrumentation/src/androidTest/java/leakcanary/Profiler.kt

```diff
@@ -0,0 +1,60 @@
+package leakcanary
+
+import android.util.Log
+
+/**
+ * Helper class for working with Android Studio's Profiler
+ */
+internal object Profiler {
+  private const val SLEEP_TIME = 1000L
+  private const val SAMPLING_THREAD_NAME = "Sampling Profiler"
+  private const val TAG = "Profiler"
+
+  /**
+   * Wait until Profiler is attached and CPU Sampling is started.
+   * Calling this on main thread can lead to ANR if you try to interact with UI while it's waiting for
+   * profiler.
+   * Note: only works with 'Sample Java Methods' profiling, won't work with 'Trace Java Methods'!
+   */
+  fun waitForSamplingStart() {
+    Log.d(TAG, "Waiting for sampling to start. Go to Profiler -> CPU -> Record")
+    sleepUntil { samplingThreadExists() }
+    Thread.sleep(SLEEP_TIME) //Wait a bit more to ensure profiler started sampling
+    Log.d(TAG, "Sampling started! Proceeding...")
```

Comment:
Why Log.d not SharkLog?

Reply:
I was thinking in terms of publicly available utility instead of LeakCanary's specific. Will definitely change it to SharkLog!